lib/maze/schemas/trace_validator.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require_relative '../helper'
+
+module Maze
+  module Schemas
+
+    # Contains a set of pre-defined validations for ensuring traces are correct
+    class TraceValidator
+
+      # Whether the trace passed the validation, one of true, false, or nil (not run)
+      #   @returns [Boolean|nil] Whether the validation was successful
+      attr_reader :success
+      attr_reader :errors
+
+      # Creates the validator
+      #
+      #   @param body [Hash] The body of the trace to validate
+      def initialize(body)
+        @body = body
+        @success = nil
+        @errors = []
+      end
+
+      # Runs the validation against the trace given
+      def validate
+        @success = true
+        regex_comparison('resourceSpans.0.scopeSpans.0.spans.0.spanId', '^[A-Fa-f0-9]{16}$')
+        regex_comparison('resourceSpans.0.scopeSpans.0.spans.0.traceId', '^[A-Fa-f0-9]{32}$')
+        regex_comparison('resourceSpans.0.scopeSpans.0.spans.0.kind', '^[A-Za-z_]+')
```

Comment:
`kind` is supposed to be an integer: 

> Values of enum fields MUST be encoded as integer values. Unlike the standard Protobuf JSON Mapping, which allows values of enum fields to be encoded as either integer values or as enum name strings, only integer enum values are allowed in OTLP JSON Protobuf Encoding; the enum name strings MUST NOT be used. For example, the kind field with a value of SPAN_KIND_SERVER in a Span can be represented like this: { “kind”: 2, ... }
>
> https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md#json-protobuf-encoding

The mapping is:

```
Unspecified = 0
Internal = 1
Server = 2
Client = 3
Producer = 4
Consumer = 5
```

So a valid `kind` is an integer `>= 0 && <= 5`